Zend/zend_object_handlers.h

```diff
@@ -205,7 +205,7 @@ ZEND_API zval *zend_std_get_static_property_with_info(zend_class_entry *ce, zend
 ZEND_API zval *zend_std_get_static_property(zend_class_entry *ce, zend_string *property_name, int type);
 ZEND_API ZEND_COLD bool zend_std_unset_static_property(zend_class_entry *ce, zend_string *property_name);
 ZEND_API zend_function *zend_std_get_constructor(zend_object *object);
-ZEND_API struct _zend_property_info *zend_get_property_info(zend_class_entry *ce, zend_string *member, int silent);
+ZEND_API struct _zend_property_info *zend_get_property_info(const zend_class_entry *ce, zend_string *member, int silent);
```

Comment:
I suppose the reason why the ``zend_string`` is not marked as ``const`` is because the hash can get set?

Reply:
Yes. Didn't compile.
I wonder whether we make it `const` anyway and then cast away the const when the hash needs to be updated, to signal via the declared API that the actual string value will not be modified by the function.
(C++ has the `mutable` keyword for ugly real-world mess like that.)

Reply:
My understanding was that actually modifying a ``const`` parameter by casting away the const is UB? As the content may be stored in read-only memory on some platforms.

But then my understanding of the C spec is very limited.

Reply:
It's UB if the pointed-to object was initially declared `const` (which allows the compiler to make assumptions, including caching values forever and putting it in `.rodata`), which is not true for any `zend_string`.

Reply:
Okay, then maybe do that in another PR while adding such a comment in the implementation? :)